be/src/vec/exec/format/orc/vorc_reader.cpp

```diff
@@ -284,6 +301,44 @@ Status OrcReader::_init_read_columns() {
     return Status::OK();
 }
 
+void OrcReader::_init_orc_cols(const orc::Type& type, std::vector<std::string>& orc_cols,
+                               std::vector<std::string>& orc_cols_lower_case) {
+    for (int i = 0; i < type.getSubtypeCount(); ++i) {
+        orc_cols.emplace_back(type.getFieldName(i));
+        orc_cols_lower_case.emplace_back(_get_field_name_lower_case(&type, i));
+        const orc::Type* sub_type = type.getSubtype(i);
+        if (sub_type->getKind() == orc::TypeKind::STRUCT) {
+            _init_orc_cols(*sub_type, orc_cols, orc_cols_lower_case);
+        }
+    }
+}
+
+bool OrcReader::_check_acid_schema(const orc::Type& type) {
```

Comment:
How about let FE tell BE it is a torc or not? So that you don't need to check this for every file?

Reply:
Good idea. It will be improved in the full acid implementation.